packages/sanity/src/core/form/store/formState.ts

```diff
@@ -791,7 +791,7 @@ function prepareArrayOfPrimitivesInputState<T extends (boolean | string | number
     schemaType: props.schemaType,
     focused: isEqual(props.path, props.focusPath),
     focusPath: trimChildPath(props.path, props.focusPath),
-    path: props.path,
+    path: props.path.concat([0]),
```

Comment:
This is is what fixes the broken path lines on the array items 
<img width="984" alt="Screenshot 2023-07-13 at 02 45 46" src="https://github.com/sanity-io/sanity/assets/6951139/621b09f6-ea18-4851-8ee2-100158481574">

I was trying to find a neater way (like we have for the review changes in the `BuildChangeList` which actually builds the right paths) of setting this up since I don't know if there is the possibility for complications for "prepareArrayOfPrimitivesInputState" specifically (I don't think that there will be since it doesn't go any "deeper" like into a popover or something) but any pointers would be swell 🥲